.github/workflows/dev-build.yml

```
@@ -76,18 +76,16 @@ jobs:
         with:
           node-version: "12"
 
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-      - uses: actions/cache@v2.1.3
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      - name: Cache node_modules
+        uses: actions/cache@v2.1.3
+        id: cached-node_modules
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          path: |
+            node_modules
```

Think this might need a glob for the sub-node_modules
```suggestion
           **/node_modules
```

---

Nope, it's not necessary. All tests pass when I trigger a re-build (which clearly skips the `yarn install` because it's cache hit). 
I.e. non of the tests fail when things like `client/node_modules` is not present.

---

There is also the node_modules in the `deploy` even if Yarn doesn't push into the `client` folder in my local right now, I've seen it appear sometimes

---

But that doesn't get used by *this* `yarn install`. There's a separate `yarn install` for the node within `deployer`.

---

but the way the hash is created, it's using all `yarn.lock` files, so they should both be using this cache, or the hash could be adjusted to just use the root lock file

---

Good catch! Let's make that match then.

---

Updated. r?

---

LGTM, if this is just for the root. Might want to add one to the deployer using the other yarn.lock path.

Not using Yarn for much, but I'm not sure if there is a difference with this vs https://github.com/actions/cache/blob/main/examples.md#node---yarn which I think caches a different folder, but always runs the install, but should be fast because of the local caching

---

In general, I find that that pattern is slower. 
That pattern has an advantage that you dont need to put the `if: ...` as we do here. 
Also, caching the downloads is a better caching strategy if the `yarn.lock` changes a lot. Then you'd get like 95% cache hits (but only on the downloads) each time. In our case, it's either 0% or 100% cache hit. And if `yarn.lock` was to change a lot, over time, you'd get 0% cache hit, 0% cache hit, 0% cache hit, etc. 

The reason it's so much faster to cache the whole node_modules must be because it's not only saving you from doing network downloads, but also all the other Node I/O work that has to be done to unpack the tarballs.